score-lang-compiler/src/main/java/org/openscore/lang/compiler/modeller/transformers/ForTransformer.java

```diff
@@ -30,22 +33,45 @@ public ForLoopStatement transform(String rawData) {
             return null;
         }
 
-        Pattern regex = Pattern.compile(FOR_REGEX);
-        Matcher matcher = regex.matcher(rawData);
+        ForLoopStatement forLoopStatement;
         String varName;
         String collectionExpression;
-        if (matcher.find()) {
-            varName = matcher.group(2);
-            collectionExpression = matcher.group(4);
+
+        Pattern regexSimpleFor = Pattern.compile(FOR_REGEX);
+        Matcher matcherSimpleFor = regexSimpleFor.matcher(rawData);
+
+        if (matcherSimpleFor.find()) {
+            // case: value in variable_name
+            varName = matcherSimpleFor.group(2);
+            collectionExpression = matcherSimpleFor.group(4);
+            forLoopStatement = new ForLoopStatement(varName, collectionExpression, ForLoopStatement.Type.LIST);
         } else {
-            varName = StringUtils.substringBefore(rawData, FOR_IN_KEYWORD);
-            collectionExpression = StringUtils.substringAfter(rawData, FOR_IN_KEYWORD);
-        }
-        varName = varName.trim();
-        if (isContainInvalidChars(varName)) {
-            throw new RuntimeException("for loop var name cannot contain invalid chars");
+            String beforeInKeyword = StringUtils.substringBefore(rawData, FOR_IN_KEYWORD);
+            collectionExpression = StringUtils.substringAfter(rawData, FOR_IN_KEYWORD).trim();
+
+            Pattern regexKeyValueFor = Pattern.compile(KEY_VALUE_PAIR_REGEX);
+            Matcher matcherKeyValueFor = regexKeyValueFor.matcher(beforeInKeyword);
+
+            if (matcherKeyValueFor.find()) {
+                // case: (key value)
+                String keyName = matcherKeyValueFor.group(2);
+                String valueName = matcherKeyValueFor.group(4);
```

Comment:
I think that if we have here the key and the value separated, it is a shame that we concat them with and a comma and then split them again at runtime.
I would prefer having to different constructors to the `ForLoopStatement` or having some subclass with the additional map data. 

and by that we can also not enforce the user to pass the `Type` eachtime
WDYT?